File: internal/plan/output.go
package plan

import (
	"path/filepath"
	"strings"
)

type OutputValues struct {
	Lang     string
	RelPath  string
	BaseName string
	Ext      string
}

func ExpandOutput(template string, values OutputValues) string {
	out := template
	out = strings.ReplaceAll(out, "{lang}", values.Lang)
	out = strings.ReplaceAll(out, "{relpath}", filepath.ToSlash(values.RelPath))
	out = strings.ReplaceAll(out, "{basename}", values.BaseName)
	out = strings.ReplaceAll(out, "{ext}", values.Ext)
	out = filepath.FromSlash(out)
	return filepath.Clean(out)
}

File: src/output.ts
import { normalize, sep } from "path";

export interface OutputValues {
  lang: string;
  relpath: string;
  basename: string;
  ext: string;
}

export function expandOutput(template: string, values: OutputValues): string {
  let out = template;
  out = out.replaceAll("{lang}", values.lang);
  out = out.replaceAll("{relpath}", values.relpath.replaceAll(sep, "/"));
  out = out.replaceAll("{basename}", values.basename);
  out = out.replaceAll("{ext}", values.ext);
  // Convert forward slashes to OS path separators, then normalize
  out = out.replaceAll("/", sep);
  return normalize(out);
}
